client/src/app/pages/asset-generators/asset-generators.tsx

```diff
@@ -0,0 +1,444 @@
+  );
+
+  const { mutate: deleteGenerator } = useDeleteGeneratorMutation(
+    onDeleteSuccess,
+    onError
+  );
+
+  const columnNames = useMemo(
+    () => ({
+      name: t(COLUMN_NAMES.name),
+      repository: t(COLUMN_NAMES.repository),
+      parameters: t(COLUMN_NAMES.parameters),
+      values: t(COLUMN_NAMES.values),
+    }),
+    [t]
+  );
+
+  const filterCategories = useMemo(
+    () => [
+      {
+        categoryKey: "name",
+        title: t("terms.name"),
+        type: FilterType.search,
+        placeholderText:
+          t("actions.filterBy", {
+            what: t("terms.name").toLowerCase(),
+          }) + "...",
+        getItemValue: (generator: AssetGenerator) => {
+          return generator?.name ?? "";
+        },
+      },
+      {
+        categoryKey: "repository",
+        title: t("terms.repository"),
+        type: FilterType.search,
+        placeholderText:
+          t("actions.filterBy", {
+            what: t("terms.repository").toLowerCase(),
+          }) + "...",
+        getItemValue: (generator: AssetGenerator) => {
+          return generator?.repository?.url ?? "";
+        },
+      },
+    ],
+    [t]
+  );
+
+  const getSortValues = useCallback(
+    (generator: AssetGenerator) => ({
+      name: generator.name ?? "",
+      repository: generator.repository?.url ?? "",
+    }),
+    []
+  );
+
+  const tableControls = useLocalTableControls({
+    tableName: "generators-table",
+    persistTo: "urlParams",
+    persistenceKeyPrefix: TablePersistenceKeyPrefix.generators,
+    idProperty: "id",
+    dataNameProperty: "name",
+    items: generators || [],
+    isLoading: isFetching,
+    hasActionsColumn: true,
+    columnNames,
+    isFilterEnabled: true,
+    isSortEnabled: true,
+    isPaginationEnabled: true,
+    isActiveItemEnabled: true,
+    filterCategories,
+    sortableColumns: ["name", "repository"],
+    getSortValues,
+    initialSort: { columnKey: "name", direction: "asc" },
+  });
+
+  const {
+    currentPageItems,
+    numRenderedColumns,
+    propHelpers: {
+      toolbarProps,
+      filterToolbarProps,
+      paginationToolbarItemProps,
+      paginationProps,
+      tableProps,
+      getThProps,
+      getTrProps,
+      getTdProps,
+    },
+    activeItemDerivedState: { activeItem, clearActiveItem },
+  } = tableControls;
+
+  const handleCreateGenerator = useCallback(() => {
+    setOpenCreateGenerator(true);
+  }, []);
+
+  const handleCloseCreateGenerator = useCallback(() => {
+    setOpenCreateGenerator(false);
+  }, []);
+
+  const handleEditGenerator = useCallback((generator: AssetGenerator) => {
+    setGeneratorToEdit(generator);
+  }, []);
+
+  const handleCloseEditGenerator = useCallback(() => {
+    setGeneratorToEdit(null);
+  }, []);
+
+  const handleDeleteGenerator = useCallback((generator: AssetGenerator) => {
+    setGeneratorToDelete(generator);
+  }, []);
+
+  const handleCancelDelete = useCallback(() => {
+    setGeneratorToDelete(null);
+  }, []);
```

Comment:
Why have all of these in a callback? The set functions for a useState hook should already be stable.